tests/test_harvester_global_ulwrf.py

```diff
@@ -0,0 +1,225 @@
+                        'bfg_1994010106_fhr06_ulwrf_avetoa_control.nc',
+                        'bfg_1994010106_fhr09_ulwrf_avetoa_control.nc',
+                        'bfg_1994010112_fhr06_ulwrf_avetoa_control.nc',
+                        'bfg_1994010112_fhr09_ulwrf_avetoa_control.nc',
+                        'bfg_1994010118_fhr06_ulwrf_avetoa_control.nc',
+                        'bfg_1994010118_fhr09_ulwrf_avetoa_control.nc',
+                        'bfg_1994010200_fhr06_ulwrf_avetoa_control.nc']
+
+DATA_DIR = os.path.join(Path(__file__).parent.parent.resolve(), 'data')
+GRIDCELL_AREA_DATA_PATH = os.path.join(DATA_DIR,
+                                       'gridcell-area' +
+                                       '_noaa-ufs-gefsv13replay-pds' +
+                                       '_bfg_control_1536x768_20231116.nc')
+CONFIGS_DIR = 'configs'
+PYTEST_CALLING_DIR = Path(__file__).parent.resolve()
+TEST_DATA_PATH = os.path.join(PYTEST_CALLING_DIR, 'data')
+BFG_PATH = [os.path.join(TEST_DATA_PATH,
+                         file_name) for file_name in TEST_DATA_FILE_NAMES]
+
+VALID_CONFIG_DICT = {'harvester_name': hv_registry.DAILY_BFG,
+                     'filenames' : BFG_PATH,
+                     'statistic': ['mean','variance', 'minimum', 'maximum'],
+                     'variable': ['ulwrf_avetoa']}
+
+def test_gridcell_area_conservation(tolerance=0.001):
+
+    gridcell_area_data = Dataset(GRIDCELL_AREA_DATA_PATH)
+    
+    assert gridcell_area_data['area'].units == 'steradian'
+    
+    sum_gridcell_area = np.sum(gridcell_area_data.variables['area'])
+    
+    assert sum_gridcell_area < (1 + tolerance) * 4 * np.pi
+    assert sum_gridcell_area > (1 - tolerance) * 4 * np.pi
+    
+    gridcell_area_data.close()
+
+def test_variable_names():
+    data1 = harvest(VALID_CONFIG_DICT)
+    assert data1[0].variable == 'ulwrf_avetoa'
+
+def test_global_mean_values(tolerance=0.001):
+    """ 
+        The value of 242.84720151427342 the mean value of the 
+        global meas calculated from these eight forecast files:
+
+        bfg_1994010100_fhr09_ulwrf_avetoa_control.nc
+        bfg_1994010106_fhr06_ulwrf_avetoa_control.nc
+        bfg_1994010106_fhr09_ulwrf_avetoa_control.nc
+        bfg_1994010112_fhr06_ulwrf_avetoa_control.nc
+        bfg_1994010112_fhr09_ulwrf_avetoa_control.nc
+        bfg_1994010118_fhr06_ulwrf_avetoa_control.nc
+        bfg_1994010118_fhr09_ulwrf_avetoa_control.nc
+        bfg_1994010200_fhr06_ulwrf_avetoa_control.nc
+        
+        When averaged together, these files represent a 24 hour mean. The 
+        average value hard-coded in this test was calculated from 
+        forecast files using a separate python code.
+    """
+    data1 = harvest(VALID_CONFIG_DICT)
+    global_mean = 242.84720151427342 
+    assert data1[0].value <= (1 + tolerance) * global_mean
+    assert data1[0].value >= (1 - tolerance) * global_mean
+ 
+def test_global_mean_values2(tolerance=0.001):
```

Comment:
small detail but in best practices I believe this test should have a more descriptive name than just adding "2" to distinguish what the differences between the two tests are. It. appears to me that one is a provided value and one a calculated value which is a fairly different test setup even if its testing the same result value (i.e. it's not necessarily the same code with different inputs which is when I would think adding a 2 on a test name would be more appropriate). Could we add a descriptive word to this test name? perhaps calculated?

Reply:
updated: now test_global_mean_values_offline() and test_global_mean_values_netCDF4()